### DISCUSSION TITLE
from_json/to_json unnamed namespace in .cpp

### DISCUSSION THREAD

**Discussion by lglgdouble** [nlohmann/json#3813] [Q&A]
namespace xdrcfg {
struct RcaReqHttpMsg {
};
// A
namespace {
void to_json(nlohmann::json &j, const RcaReqHttpMsg &p) {
}

void from_json(const nlohmann::json &j, RcaReqHttpMsg &p) {
}
}
}

// or B
static void to_json(nlohmann::json &j, const RcaReqHttpMsg &p) {
}

static void from_json(const nlohmann::json &j, RcaReqHttpMsg &p) {
}
}

RcaReqHttpMsg l_req;
js.get_to(l_req);

 B is ok, why?

**Comment by falbrechtskirchinger** [comment]
The library looks for the `to_json`/`from_json` functions in the namespace of the type. The functions in the anonymous namespace aren't considered because the type `RcaReqHttpMsg` resides in a different namespace.
